test/WebSites/InlineConstraintsWebSite/Constraints/ItemTypeConstraint.cs

```diff
@@ -0,0 +1,42 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNet.Http;
+using Microsoft.AspNet.Routing;
+
+namespace InlineConstraintsWebSite.Constraints
+{
+    public class ItemTypeConstraint : IRouteConstraint
```

Comment:
A classic one is something like ISBN10 and ISBN13. It's just a line or two of boring math to compute, and showing how to support both ISBN formats could be neat.